packages/ui-app/src/Params/Param/Balance.tsx

```diff
@@ -0,0 +1,192 @@
+// Copyright 2017-2018 @polkadot/ui-app authors & contributors
+// This software may be modified and distributed under the terms
+// of the ISC license. See the LICENSE file for details.
+
+import { ApiProps } from '@polkadot/ui-react-rx/types';
+import { I18nProps } from '../../types';
+import { Props as BareProps } from '../types';
+
+import BN from 'bn.js';
+import React from 'react';
+import withApi from '@polkadot/ui-react-rx/with/api';
+
+import isValidBalance from '../../util/isValidBalance';
+import Bare from './Bare';
+import InputNumber from '../../InputNumber';
+import translate from '../../translate';
+
+type Props = I18nProps & ApiProps & BareProps;
+
+type State = {
+  error?: React.ReactNode,
+  info?: React.ReactNode
+};
+
+class Balance extends React.PureComponent<Props, State> {
+  state: State;
+
+  constructor (props: Props) {
+    super(props);
+
+    this.state = {
+      error: '',
+      info: ''
+    };
+  }
+
+  render () {
+    const { className, defaultValue: { value }, isError, label, style, t, withLabel } = this.props;
+    const { error, info } = this.state;
+    const defaultValue = new BN((value as BN).toString(10) || '0').toString(10);
+
+    const maxLengthForLatestChainSpec = 39;
+
+    return (
+      <Bare
+        className={className}
+        style={style}
+      >
+        <InputNumber
+          className='large'
+          defaultValue={defaultValue || '0'}
+          error={error}
+          info={info}
+          isError={isError}
+          label={label}
+          maxLength={maxLengthForLatestChainSpec}
+          onChange={this.onChange}
+          onKeyDown={this.onKeyDown}
+          onKeyUp={this.onKeyUp}
+          placeholder={t('account.balance.placeholder', {
+            defaultValue: 'Between 1 testnet DOT and the available testnet DOT balance (minus 1) of the account'
```

Comment:
We should no mention testnet, keep it generic otherwise we need to come back to this. Additionally, we can allow 0 as an input. (Just not negative). Why "minus 1"

Reply:
Ok. I can't recall the reason for the "minus 1". It may have been because of errors in the Polkadot Node logs when I tried sending the full account balance, but I'll have to check again once I revive Alice.
Why would we allow them to enter 0 as an input?